commands/service_profile_init.go

```diff
@@ -0,0 +1,114 @@
+// This file is part of arduino-cli.
+//
+// Copyright 2025 ARDUINO SA (http://www.arduino.cc/)
+//
+// This software is released under the GNU General Public License version 3,
+// which covers the main part of arduino-cli.
+// The terms of this license can be found at:
+// https://www.gnu.org/licenses/gpl-3.0.en.html
+//
+// You can be released from the requirements of the above licenses by purchasing
+// a commercial license. Buying such a license is mandatory if you want to
+// modify or otherwise use the software for commercial activities involving the
+// Arduino software without disclosing the source code of your own applications.
+// To purchase a commercial license, send an email to license@arduino.cc.
+
+package commands
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+
+	"github.com/arduino/arduino-cli/commands/cmderrors"
+	"github.com/arduino/arduino-cli/commands/internal/instances"
+	"github.com/arduino/arduino-cli/internal/arduino/sketch"
+	"github.com/arduino/arduino-cli/internal/i18n"
+	"github.com/arduino/arduino-cli/pkg/fqbn"
+	rpc "github.com/arduino/arduino-cli/rpc/cc/arduino/cli/commands/v1"
+	"github.com/arduino/go-paths-helper"
+)
+
+// InitProfile creates a new project file if it does not exist. If a profile name with the associated FQBN is specified,
+// it is added to the project.
+func (s *arduinoCoreServerImpl) InitProfile(ctx context.Context, req *rpc.InitProfileRequest) (*rpc.InitProfileResponse, error) {
+	sketchPath := paths.New(req.GetSketchPath())
+	projectFilePath, err := sketchPath.Join("sketch.yaml").Abs()
+	if err != nil {
+		return nil, err
+	}
+
+	// Returns an error if the main file is missing from the sketch so there is no need to check if the path exists
+	sk, err := sketch.New(sketchPath)
+	if err != nil {
+		return nil, err
+	}
+
+	if !projectFilePath.Exist() {
+		err := projectFilePath.WriteFile([]byte("profiles:\n"))
+		if err != nil {
+			return nil, err
+		}
+	}
```

Comment:
In the GRPC api, is possible to endup with a project file containing an invalid yaml:
```yaml
profiles:

```
This can happen if: the path doesn't exists and the ProfileName is empty

```suggestion
	if !projectFilePath.Exist() && req.GetProfileName() == ""{
	// return error
	    return nil, error 
	}

	if !projectFilePath.Exist() {
		err := projectFilePath.WriteFile([]byte("profiles:\n"))
		if err != nil {
			return nil, err
		}
	}
```

Reply:
Resolved, when no profile is specified the resulting `sketch.yaml` is:
`profiles: {}`